examples/fewshot_protonet/main.py

```diff
@@ -0,0 +1,126 @@
+    return args
+
+
+def main():
+    # ---- get args ----
+    args = arg_parse()
+
+    # ---- get configurations ----
+    cfg_path = args.cfg
+    cfg = get_cfg_defaults()
+    cfg.merge_from_file(cfg_path)
+    cfg.freeze()
+
+    # ---- set model ----
+    net = eval(f"{cfg.MODEL.BACKBONE}(weights={cfg.MODEL.PRETRAIN_WEIGHTS})")
+    if cfg.MODEL.BACKBONE.startswith("resnet"):
+        net.fc = Flatten()
+
+    model = ProtoNetTrainer(
+        net=net,
+        train_num_classes=cfg.TRAIN.NUM_CLASSES,
+        train_num_support_samples=cfg.TRAIN.NUM_SUPPORT_SAMPLES,
+        train_num_query_samples=cfg.TRAIN.NUM_QUERY_SAMPLES,
+        val_num_classes=cfg.VAL.NUM_CLASSES,
+        val_num_support_samples=cfg.VAL.NUM_SUPPORT_SAMPLES,
+        val_num_query_samples=cfg.VAL.NUM_QUERY_SAMPLES,
+        devices=cfg.DEVICE,
+        optimizer=cfg.TRAIN.OPTIMIZER,
+        lr=cfg.TRAIN.LEARNING_RATE,
+    )
+
+    # ---- set data loader ----
+    transform = get_transform(kind="few-shot", augment=False)
+    train_set = NWayKShotDataset(
+        path=cfg.DATASET.ROOT,
+        mode="train",
+        num_support_samples=cfg.TRAIN.NUM_SUPPORT_SAMPLES,
+        num_query_samples=cfg.TRAIN.NUM_QUERY_SAMPLES,
+        transform=transform,
+    )
+    train_dataloader = DataLoader(train_set, batch_size=cfg.TRAIN.NUM_CLASSES, shuffle=True, drop_last=True)
+    val_set = NWayKShotDataset(
+        path=cfg.DATASET.ROOT,
+        mode="val",
+        num_support_samples=cfg.VAL.NUM_SUPPORT_SAMPLES,
+        num_query_samples=cfg.VAL.NUM_QUERY_SAMPLES,
+        transform=transform,
+    )
+    val_dataloader = DataLoader(val_set, batch_size=cfg.VAL.NUM_CLASSES, drop_last=True)
+    test_set = NWayKShotDataset(
+        path=cfg.DATASET.ROOT,
+        mode="test",
+        num_support_samples=cfg.VAL.NUM_SUPPORT_SAMPLES,
+        num_query_samples=cfg.VAL.NUM_QUERY_SAMPLES,
+        transform=transform,
+    )
+    test_dataloader = DataLoader(test_set, batch_size=cfg.VAL.NUM_CLASSES, drop_last=True)
+
+    # ---- set logger ----
+    experiment_time = datetime.now().strftime("%Y-%m-%d-%H-%M-%S")
+    logger = pl.loggers.TensorBoardLogger(cfg.OUTPUT.OUT_DIR, name=experiment_time)
+    logger.log_hyperparams(cfg)
+
+    # ---- set callbacks ----
+    dirpath = os.path.join(cfg.OUTPUT.OUT_DIR, experiment_time, cfg.OUTPUT.WEIGHT_DIR)
+    model_checkpoint = pl.callbacks.ModelCheckpoint(
+        dirpath=dirpath,
+        filename="{epoch}-{val_acc:.2f}",
+        monitor="val_acc",
+        mode="max",
+        save_top_k=cfg.OUTPUT.SAVE_TOP_K,
+        save_last=cfg.OUTPUT.SAVE_LAST,
+        verbose=True,
+    )
+
+    # ---- set trainer ----
+    trainer = pl.Trainer(
+        devices=args.devices,
+        max_epochs=cfg.TRAIN.EPOCHS,
+        logger=logger,
+        callbacks=[model_checkpoint],
+        accelerator="gpu" if args.devices > 0 else "cpu",
+        log_every_n_steps=cfg.OUTPUT.SAVE_FREQ,
+    )
+
+    # ---- training ----
```

Comment:
But you have both fit and test below so this block is not just training.

Reply:
Changed